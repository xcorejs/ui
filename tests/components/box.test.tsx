import 'jest-styled-components';

import { Box } from '@xcorejs/ui';
import React from 'react';
import renderer from 'react-test-renderer';

test('Box component', () => {
  const component = renderer.create(
    <>
      <Box>Empty</Box>
      <Box
        background="crimson"
        color="white"
        p="10px"
        my="15px"
      >
        Box 1
      </Box>
      <Box
        background="navy"
        color="white"
        p="10px"
        my="15px"
        _hover={{
          background: { _: 'crimson', md: 'green' }
        }}
      >
        Box 2
      </Box>
      <Box
        background="navy"
        color="white"
        p="10px"
        my="15px"
        _before={{
          content: '"before"',
          background: 'green'
        }}
        _after={{
          content: '"after"',
          background: 'green'
        }}
      >
        Box 3
      </Box>

      <Box role="group">
        <Box>Outside</Box>
        <Box _group={{ _hover: { background: 'navy' } }}>Inside</Box>
      </Box>
    </>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
