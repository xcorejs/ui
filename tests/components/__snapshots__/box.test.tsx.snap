// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component 1`] = `
Array [
  @media screen and (min-width:64em) {

}

<div
    className=""
  >
    Empty
  </div>,
  .c0 {
  padding: 10px;
  margin-top: 15px;
  margin-bottom: 15px;
  background: crimson;
  color: white;
}

@media screen and (min-width:64em) {

}

<div
    className="c0"
    color="white"
  >
    Box 1
  </div>,
  .c0 {
  padding: 10px;
  margin-top: 15px;
  margin-bottom: 15px;
  background: navy;
  color: white;
}

.c0:hover {
  background: crimson;
}

@media screen and (min-width:64em) {
  .c0:hover {
    background: green;
  }
}

<div
    className="c0"
    color="white"
  >
    Box 2
  </div>,
  .c0 {
  padding: 10px;
  margin-top: 15px;
  margin-bottom: 15px;
  background: navy;
  color: white;
}

.c0::before {
  content: "before";
  background: green;
}

.c0::after {
  content: "after";
  background: green;
}

@media screen and (min-width:64em) {

}

<div
    className="c0"
    color="white"
  >
    Box 3
  </div>,
  [role=group]:hover .c0 {
  background: navy;
}

@media screen and (min-width:64em) {

}

<div
    className=""
    role="group"
  >
    <div
      className=""
    >
      Outside
    </div>
    <div
      className="c0"
    >
      Inside
    </div>
  </div>,
]
`;
